packages/devextreme/js/__internal/scheduler/__tests__/__mock__/model/popup.ts

```diff
@@ -0,0 +1,136 @@
+
+    const input = this.element.querySelector(`input#${forId}`) as HTMLInputElement;
+
+    if (!input) {
+      throw new Error(`Input with id "${forId}" not found`);
+    }
+
+    return input;
+  };
+
+  setInputValueByLabel = (labelText: string, value: string): HTMLInputElement => {
+    const input = this.getInputByLabel(labelText);
+    if (!input) {
+      throw new Error(`Input with label "${labelText}" not found`);
+    }
+    input.value = '';
+
+    value.split('').forEach((char) => {
+      input.value += char;
+      input.dispatchEvent(new Event('input', { bubbles: true }));
+      input.dispatchEvent(new KeyboardEvent('keydown', { bubbles: true, key: char }));
+      input.dispatchEvent(new KeyboardEvent('keypress', { bubbles: true, key: char }));
+      input.dispatchEvent(new KeyboardEvent('keyup', { bubbles: true, key: char }));
+    });
+
+    input.dispatchEvent(new Event('change', { bubbles: true }));
+    input.dispatchEvent(new Event('input', { bubbles: true }));
+
+    return input;
+  };
+
+  getSwitchByName = (name: string): HTMLInputElement => {
+    const hiddenInput = this.element.querySelector<HTMLInputElement>(`input[name=${name}]`);
+
+    if (!hiddenInput) {
+      throw new Error(`Switch with name "${name}" not found`);
+    }
+
+    return hiddenInput;
+  };
+
+  selectRadio = (value: string): Element | null => {
+    const group = this.element.querySelector('[role="radiogroup"]');
+    if (!group) throw new Error('Radiogroup not found');
+
+    const radios = Array.from(group.querySelectorAll('[role="radio"]'));
+
+    const target = radios.find((radio) => {
+      const label = radio.getAttribute('aria-label')?.trim();
+      const text = radio.textContent?.trim();
+      return label === value || text === value;
+    });
+
+    if (!target) throw new Error(`Radio with value "${value}" not found`);
+
+    radios.forEach((r) => {
+      r.setAttribute('aria-checked', 'false');
+      r.classList.remove('dx-item-selected', 'dx-radiobutton-checked');
+    });
+
+    target.setAttribute('aria-checked', 'true');
+    target.classList.add('dx-item-selected', 'dx-radiobutton-checked');
+
+    target.dispatchEvent(new MouseEvent('click', { bubbles: true }));
+
+    return target;
+  };
+
+  getSelectedRadio = (): HTMLElement | null => this.element.querySelector('[role="radio"][aria-checked="true"]');
+
+  getSelectedRadioValue = (): string | null => {
+    const selected = this.getSelectedRadio();
+    return selected?.getAttribute('aria-label') ?? selected?.textContent?.trim() ?? null;
+  };
+
+  getForm = (): HTMLElement | null => this.element.querySelector('.dx-form');
+
+  getTitle = (): HTMLElement | null => document.querySelector('.dx-popup-title');
+
+  getDoneButton = (): HTMLButtonElement => {
+    const doneButton = this.element.querySelector('.dx-button.dx-popup-done') as HTMLButtonElement;
+    if (!doneButton) {
+      throw new Error('Done button not found');
+    }
+    return doneButton;
```

Comment:
I understand why in `selectRadio` we throw an error, because the last part of the function dependent on element existing. So it's not about the test, but test utils.

In this case we have not throw an error because it just return value. Let test writer decide what to do with this. For example, `getSelectedRadioValue` return null without error, why this function throw an error?

I think better not throw error on return value.